front/src/applications/operationalStudies/views/Scenario/components/ManageTimetableItem/Itinerary/hooks/usePathStepsMetadata.ts

```diff
@@ -0,0 +1,247 @@
+      }
+      return acc;
+    }, []);
+
+    if (opIds.length === 0) return [];
+
+    return pathStepsOperationalPoints.reduce<{ uic: number }[]>((acc, op) => {
+      const uic = op.extensions?.identifier?.uic;
+      if (uic && opIds.includes(op.id)) {
+        acc.push({ uic });
+      }
+      return acc;
+    }, []);
+  }, [pathSteps, pathStepsOperationalPoints]);
+
+  const { currentData: allOpIdsOperationalPoints } =
+    osrdEditoastApi.endpoints.postInfraByInfraIdMatchOperationalPoints.useQuery(
+      uicPayload.length > 0
+        ? {
+            infraId,
+            body: {
+              operational_point_references: uicPayload,
+            },
+          }
+        : skipToken
+    );
+
+  // 3. Merge both operational points lists to have all possible matches
+  const allOps = useMemo(() => {
+    if (
+      !allOpIdsOperationalPoints?.related_operational_points ||
+      allOpIdsOperationalPoints.related_operational_points.length === 0
+    ) {
+      return pathStepsOperationalPoints;
+    }
+    return [
+      ...pathStepsOperationalPoints,
+      ...allOpIdsOperationalPoints.related_operational_points.flat(),
+    ];
+  }, [pathStepsOperationalPoints, allOpIdsOperationalPoints]);
+
+  useEffect(() => {
+    const fetchAndSetMetadata = async () => {
+      // 4. Get all track ids of all matched operational points to get all tracks metadata
+      // and regroup ops by uic to easily get all secondary codes and track names for each path step
+      const matchedTrackIds = new Set<string>();
+      const opsByUic = new Map<number, RelatedOperationalPoint[]>();
+      allOps.forEach((op) => {
+        op.parts.forEach((part) => {
+          matchedTrackIds.add(part.track);
+        });
+        const uic = op.extensions?.identifier?.uic;
+        if (uic) {
+          if (!opsByUic.has(uic)) {
+            opsByUic.set(uic, [op]);
+          } else {
+            opsByUic.get(uic)?.push(op);
+          }
+        }
+      });
+      const allTrackIds = Array.from(matchedTrackIds);
+
+      // Add requested points track ids
+      pathSteps.forEach(
+        (step) => step.location && 'track' in step.location && allTrackIds.push(step.location.track)
+      );
+
+      const trackSectionsById = await getTrackSectionsByIds(allTrackIds);
+
+      // 5. Loop of the path steps to build the metadata map
+      const newPathStepsMetadataById = new Map<string, PathStepMetadata>();
+
+      pathSteps.forEach((pathStep) => {
+        const { location } = pathStep;
+
+        if (!location) {
+          newPathStepsMetadataById.set(pathStep.id, { isInvalid: true });
+          return;
+        }
+
+        if ('track' in location) {
+          // TODO : replace the name by the track offset label when provided by backend
+          const correspondingTrack = trackSectionsById[location.track];
+          const coordinates = getPointOnTrackCoordinates(
+            correspondingTrack?.geo,
+            correspondingTrack?.length,
```

Comment:
The `?` ca be removed 
```suggestion
            correspondingTrack.geo,
            correspondingTrack.length,
```

Reply:
Done